HM3CoreKill/HM3CoreKill/ck/HM3InGameTools.cpp

```diff
@@ -529,6 +555,50 @@ namespace ck
 				ImGui::Text("Location pointer at 0x%.8X", currentActor->ActorInformation->location);
 				ImGui::Text("Position:"); ImGui::SameLine(0.f, 4.f); ImGui::InputFloat3("", actorPosition);
 
+				{
+					if (auto boid = currentActor->m_boid; boid)
+					{
+						using BoidState = ioi::hm3::BoidState;
+
+						ImGui::Separator();
+						ImGui::Text("Human boid info (0x%.8X)", currentActor->m_boid);
+						ImGui::Spacing();
+						ImGui::Text("Boid locations : {%.4f; %.4f; %.4f} | {%.4f; %.4f; %.4f}", boid->m_pos.x, boid->m_pos.y, boid->m_pos.z, boid->m_newPos.x, boid->m_newPos.y, boid->m_newPos.z);
+						ImGui::Text("Movement speed : %.4f", boid->m_speed); 
+						
+						ImGui::Text("Boid state     : "); ImGui::SameLine(0.f, 0.f);
+						switch (boid->m_boidState)
+						{
+						case BoidState::Moving:
+							ImGui::TextColored(ImVec4(0.f, 1.f, 0.f, 1.f), "MOVE TO POINT");
+							break;
+						case BoidState::Staying:
+							ImGui::TextColored(ImVec4(0.f, 1.f, 0.f, 1.f), "STAY AT POSITION");
+							break;
+						case BoidState::ParentActorDestroyed:
+							ImGui::TextColored(ImVec4(0.f, 1.f, 0.f, 1.f), "ACTOR DEAD");
+							break;
+						case BoidState::Sitting:
+							ImGui::TextColored(ImVec4(0.f, 1.f, 0.f, 1.f), "SITTING");
+							break;
+						default:
+							ImGui::TextColored(ImVec4(1.f, 1.f, 0.f, 1.f), "UNKNOWN STATE (%d)", boid->m_boidState);
+							break;
+						}
+
+						if (ImGui::Button("Teleport test"))
+						{
+							float newPosition[] = { 344.f, -267.f, 238.f };
```

Comment:
Rename button to "teleport", add the widget for float3 input and teleport boid to passed coordinate.